src/librustc_metadata/creader.rs

```diff
@@ -452,7 +452,21 @@ impl<'a> CrateLoader<'a> {
         if dep.is_none() {
             self.used_extern_options.insert(name);
         }
-        self.maybe_resolve_crate(name, span, dep_kind, dep).unwrap_or_else(|err| err.report())
+        let mut skip_loading = false;
+        if !name.as_str().is_ascii() {
+            self.sess
+                .struct_span_warn(
```

Comment:
I meant you can emit the "cannot load a crate with a non-ascii name `foo`" error here and proceed to actually loading it for recovery.

No `skip_loading` or "maybe a missing type or module" necessary.

It's better than a generic "can't find crate" error and hard-coded warnings are discouraged.

Reply:
@petrochenkov oh i must be misunderstanding it, but in #55467 the "requirement" said that 

> Similarly to out-of-line modules (mod фоо;), extern crates and paths with a first segment naming a crate should not be able to do filesystem search using those non-ASCII identifiers (i.e. no , extern crate ьаг; or му_сгате::baz).

So i was i actually trying to to implement the "not be able to do filesystem search" part... Would you mind amending that text and clarify that a little if we still want to actually do filesystem search in this case?

Reply:
Also, the `crate_name_nonascii_forbidden-2.rs` test case seems doesn't call this `resolve_crate` function. It is calling `maybe_process_path_extern` function instead. Do we want to do something in this case? Or should we just remove the test case if we want to do nothing here?

Reply:
>should not be able to do filesystem search using those non-ASCII identifiers

Ah, all these requirements should be suffixed by "if compilation succeeds".
Once we have reported and error we can do anything for recovery - load unloadable, see invisible, etc.

Reply:
>Also, the `crate_name_nonascii_forbidden-2.rs` test case seems doesn't call this resolve_crate function.

That's because `use му_сгате::baz` in it is just an unresolved import, there's no crate named `му_сгате` anywhere in scope.
You'll need to use `--extern` for it to appear.

`maybe_process_path_extern` in this case is only called during the search for suggestion candidates.

Reply:
Thanks!